### DISCUSSION TITLE
Is it possible to enable "restore focus on unmount" functionality for Dialog?

### DISCUSSION THREAD

**Discussion by jasikpark** [ariakit/ariakit#3961] [Questions]
I have a deletion modal for items in a table built w/ the ariakit Dialog, and currently when the element that was previously focused is removed from the DOM, the root html element is restored to focus instead.

Could something like https://x.com/devongovett/status/1577375026088550402 be enabled?

See: https://react-spectrum.adobe.com/react-aria/FocusScope.html

So that instead of restoring focus to the root node, it's restored close to other nodes?

**Comment by jasikpark** [comment]
Looks like they implement it by going up FocusScope trees to find the closest one to the deleted node? So in the example, it requires adding a FocusScope to the table. https://x.com/devongovett/status/1577376104716783616

**Reply by jasikpark** [reply]
Wrapping Dialog in a react-aria FocusScope and then wrapping my Tables in one worked well. Would love to have an implementation that allows me to only use ariakit/react though!

**Comment by diegohaz** [comment]
It's not automatic, but you can choose what to focus on if the original element is inaccessible using the function version of [`autoFocusOnHide`](https://ariakit.org/reference/dialog#autofocusonhide).

Related: [Restoring focus on hide](https://ariakit.org/examples/dialog-next-router#restoring-focus-on-hide)
